### DISCUSSION TITLE
model_get_param_info do not show shader infos

### DISCUSSION THREAD

**Discussion by mvvvv** [StereoKit/StereoKit#742] [Q&A] [answered]
Hi, I've implemented model_get_param_info for rust, hoping to discover all the secrets of the shaders, but I only found some information accessible by MatParamName plus an unknown sk-cubemap texture. Shouldn't we see the specific information for the chosen shader?

what I get for 2 materials:
One create inside stereokit with png files 
```
[SK info] Material Pararms for "marble" -> 10
[SK info] --- "color" -> Some("RGBA : 1.0/1.0/1.0 : 1.0")
[SK info] --- "emission_factor" -> Some("RGBA : 0.0/0.0/0.0 : 0.0")
[SK info] --- "metallic" -> Some("0")
[SK info] --- "roughness" -> Some("0.25")
[SK info] --- "tex_scale" -> Some("1")
[SK info] --- "sk_cubemap" -> Some("default/tex")
[SK info] --- "diffuse" -> Some("textures/marble/1K_carrara_gioa_p1004___polished___marble_diffuse.png")
[SK info] --- "emission" -> Some("default/tex")
[SK info] --- "metal" -> Some("default/tex")
[SK info] --- "occlusion" -> Some("textures/marble/1K_carrara_gioa_p1004___polished___marble_ao.png")
[SK info] test get >>> Some("mobiles.gltf/tex/1")
```

The other is a procedural texture from blender with some adjustement (occlusion, roughness, color) :
```
[SK info] Material Pararms for "floor" -> 10
[SK info] --- "color" -> Some("RGBA : 0.0/0.0/0.0 : 1.0")
[SK info] --- "emission_factor" -> Some("RGBA : 0.0/0.0/0.0 : 0.0")
[SK info] --- "metallic" -> Some("0")
[SK info] --- "roughness" -> Some("0.7")
[SK info] --- "tex_scale" -> Some("1")
[SK info] --- "sk_cubemap" -> Some("default/tex")
[SK info] --- "diffuse" -> Some("default/tex")
[SK info] --- "emission" -> Some("default/tex")
[SK info] --- "metal" -> Some("mobiles.gltf/tex/1")
[SK info] --- "occlusion" -> Some("textures/marble/1K_carrara_gioa_p1004___polished___marble_ao.png")
```

**Comment by mvvvv** [comment, marked as answer]
there is no shader in a material ... I miss something sorry this message never existed ...

**Reply by maluoi** [reply]
A `Material` is a combination of a `Shader`, and a parameter buffer. You can access the `Material`'s `Shader` via `material_get_shader`, but all the really interesting bits of data are acquired via the `Material` APIs anyhow.

It's also important to note that GLTF doesn't export a wide range of shader effects either, it leans heavily on PBR! So most materials coming out of GLTF files are imported using StereoKit's PBR shader. On some occasions maybe the Unlit shader too. If you're trying to use Blender's shader graph for fancy materials, there's no good way to communicate those shader graphs with any external format. You'll need to check Blender's exporter docs to see exactly where the limits are for the format and Blender's capabilities.
